src/realm-settings/key-providers/ecdsa-generated/ECDSAGeneratedModal.tsx

```diff
@@ -0,0 +1,36 @@
+import React from "react";
+import { Modal, ModalVariant } from "@patternfly/react-core";
+import { useTranslation } from "react-i18next";
+import { ECDSAGeneratedForm } from "./ECDSAGeneratedForm";
+
+type ECDSAGeneratedModalProps = {
+  providerType: string;
+  handleModalToggle: () => void;
+  refresh: () => void;
+  open: boolean;
+};
+
+export const ECDSAGeneratedModal = ({
+  providerType,
+  handleModalToggle,
+  open,
+  refresh,
+}: ECDSAGeneratedModalProps) => {
+  const { t } = useTranslation("realm-settings");
+
+  return (
+    <Modal
+      className="add-provider-modal"
+      variant={ModalVariant.medium}
+      title={t("addProvider")}
+      isOpen={open}
+      onClose={handleModalToggle}
+    >
+      <ECDSAGeneratedForm
+        providerType={providerType!}
```

Comment:
Using the non-null assertion operator can lead to exceptions at run-time.
```suggestion
        providerType={providerType}
```